accelerator/config.c

```diff
@@ -5,10 +5,74 @@
 
 static logger_id_t logger_id;
 
-status_t ta_config_init(ta_config_t* const info, iota_config_t* const tangle,
-                        iota_client_service_t* const service) {
+struct option* cli_build_options() {
+  struct option* long_options =
+      (struct option*)malloc(cli_cmd_num * sizeof(struct option));
+  for (int i = 0; i < cli_cmd_num; ++i) {
+    long_options[i].name = ta_cli_arguments_g[i].name;
+    long_options[i].has_arg = ta_cli_arguments_g[i].has_arg;
+    long_options[i].flag = NULL;
+    long_options[i].val = ta_cli_arguments_g[i].val;
+  }
+  return long_options;
+}
+
+status_t cli_config_set(ta_config_t* const info, iota_config_t* const tangle,
+                        ta_cache_t* const cache,
+                        iota_client_service_t* const service, int key,
+                        char* const value) {
+  if (value == NULL || info == NULL || tangle == NULL || cache == NULL ||
+      service == NULL) {
+    return SC_CONF_NULL;
+  }
+  switch (key) {
+    // TA configuration
+    case TA_HOST_CLI:
+      info->host = value;
+      break;
+    case TA_PORT_CLI:
+      info->port = value;
+      break;
+    case TA_THREAD_COUNT_CLI:
+      info->thread_count = atoi(value);
```

Comment:
should we check if `*value` in the range of `int`?
Since If the converted value would be out of the range of representable values by an int, it causes undefined behavior.

source: http://www.cplusplus.com/reference/cstdlib/atoi/